ksqldb-execution/src/main/java/io/confluent/ksql/name/ColumnNames.java

```diff
@@ -17,28 +17,30 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import io.confluent.ksql.execution.expression.tree.DereferenceExpression;
+import io.confluent.ksql.execution.expression.tree.Expression;
 import io.confluent.ksql.schema.ksql.Column;
 import io.confluent.ksql.schema.ksql.LogicalSchema;
-import io.confluent.ksql.util.KsqlConstants;
 import io.confluent.ksql.util.KsqlException;
+import java.util.HashMap;
 import java.util.List;
-import java.util.OptionalInt;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
 import java.util.Set;
-import java.util.function.Supplier;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
 public final class ColumnNames {
 
   private static final String AGGREGATE_COLUMN_PREFIX = "KSQL_AGG_VARIABLE_";
-  private static final String GENERATED_ALIAS_PREFIX = "KSQL_COL_";
+  private static final String GENERATED_ALIAS_PREFIX = "KSQL_COL";
   private static final String SYNTHESISED_COLUMN_PREFIX = "KSQL_SYNTH_";
 
-  private static final Pattern GENERATED_ALIAS_PATTERN = Pattern
-      .compile(GENERATED_ALIAS_PREFIX + "(\\d+)");
+  private static final Pattern NUMBERED_COLUMN_PATTERN = Pattern.compile("(.*?)?(?:_(\\d+))?");
```

Comment:
can we have a comment explaining the regex? also instead of relying on `?:` and indexed groups we should take advantages of named groups (https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#group%28java.lang.String%29)

Reply:
Your wish....